sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.CSharp9.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+
+var x = 100; // FN
+x = 1;
+
+void TargetTypedNew()
+{
+    Decimal d = new(100f); // FN
+    d = new(2f);
+}
+
+void NativeInts(nuint param)
+{
+    param = 1; // Noncompliant
+
+    nuint zero = 0; // ignored value
+    zero = 1;
+    Foo(zero);
+
+    nint minusOne = -1; // ignored value
+    minusOne = 1;
+    Foo(minusOne);
+
+    nint one = 1; // ignored value
+    one = 2;
+    Foo(one);
+
+    nint two = 2; // Noncompliant
+    two = 3;
+    Foo(two);
+}
+
+void PatternMatch(object param)
+{
+    object a = param;
+    if (a is not null)
```

Comment:
`Not` is not implemented on our side and I think we should add a case to show this. Maybe something similar with:
```
void PatternMatch2(int a, int b)
{
    if (b is not 5)
    {
        a = 1;
    }
    else if (b is 5)
    {
        a = 2; // Compliant - FN, the parameter value is overwritten on all possible paths
    }
}
```